packages/website/docs/docs/side-menu.md

```diff
@@ -1,8 +1,142 @@
+
+<img style="max-width:500px" src="../public/img/screenshots/side_menu.png" alt="image">
+
+You can also click the drag handle in the Block Side Menu (`⠿`) to open the Drag Handle Menu.
+
+<img style="max-width:250px" src="../public/img/screenshots/drag_handle_menu.png" alt="image">
+
+## Custom Drag Handle Menu
+
+BlockNote lets you customize which items appear in the Drag Handle Menu. Have a look at the example below, in which the color picker item is replaced with a custom item that opens an alert.
+
+::: sandbox {template=react-ts}
+
+```typescript /App.tsx
+import { Block, BlockNoteEditor } from "@blocknote/core";
+import {
+  BlockNoteView,
+  createReactBlockSideMenuFactory,
+  DragHandleMenu,
+  DragHandleMenuItem,
+  RemoveBlockButton,
+  useBlockNote,
+} from "@blocknote/react";
+import "@blocknote/core/style.css";
+
+const CustomDragHandleMenu = (props: {
+  editor: BlockNoteEditor;
+  block: Block;
+  closeMenu: () => void;
+}) => {
+  return (
+    <DragHandleMenu>
+      {/*Default button to remove the block.*/}
+      <RemoveBlockButton {...props}>
+        Delete
+      </RemoveBlockButton>
+      {/*Custom item which opens an alert when clicked.*/}
+      <DragHandleMenuItem
+        closeMenu={props.closeMenu}
+        onClick={() => {
+          window.alert("Button Pressed!");
+          props.closeMenu();
+        }}>
+        Open Alert
+      </DragHandleMenuItem>
+    </DragHandleMenu>
+  );
+};
+
+export default function App() {
+  // Creates a new editor instance.
+  const editor: BlockNoteEditor = useBlockNote({
+    uiFactories: {
+      // Makes the editor instance use the custom menu.
+      blockSideMenuFactory:
+        createReactBlockSideMenuFactory(CustomDragHandleMenu),
+    },
+  });
+  // Renders the editor instance.
+  return <BlockNoteView editor = {editor}/>;
+}
+```
+
+:::
+
+Let's look at how this is done. We first need to create a custom Drag Handle Menu using a React component. This component should take the following props:
+
+```typescript
+type CustomDragHandleMenuProps = {
+  editor: BlockNoteEditor;
+  block: Block;
+  closeMenu: () => void;
+};
+const CustomDragHandleMenu = (props: CustomDragHandleMenuProps): JSX.Element => ...;
+```
+
+You can then tell BlockNote to use your custom Drag Handle Menu using the `uiFactories` option in `useBlockNote`, but you first have to turn it into a `SideMenuFactory` that uses it. Fortunately, you can easily do this using the `createReactSideMenuFactory` function:
+
+```typescript
+const editor = useBlockNote({
+  uiFactories: {
+    blockSideMenuFactory: createReactBlockSideMenuFactory(
+      CustomBlockSideMenu
+    ),
+  },
+});
+```
+
+If all this sounds confusing, you can find out more about the `uiFactories` option in [Creating Your Own UI Elements](/docs/vanilla-js#creating-your-own-ui-elements), but it's only useful if you want to use BlockNote in a non-React app.
```

Comment:
`but it's only useful if you want to use BlockNote in a non-React app.`, then let's remove this paragraph